tests/test_language_intents.py

```diff
@@ -76,16 +76,17 @@ def do_test_language_sentences(
             if not data.sentences:
                 continue
 
-            # Domain specific files (ie light_HassTurnOn.yaml) should only match
-            # sentences for the light domain.
-            if intent_schemas[file_intent]["domain"] == file_domain:
-                assert (
-                    "domain" not in data.slots
-                ), f"File {file_name} should only have sentences without a domain slot"
-            else:
-                assert (
-                    data.slots.get("domain") == file_domain
-                ), f"File {file_name} should only have sentences with a domain slot set to {file_domain}"
+            if file_domain != "homeassistant":
```

Comment:
I'm surprised this is needed, as that is what we checked more-or-less with 

```
if intent_schemas[file_intent]["domain"] == file_domain:
```

It checks if the integration that defines the intent is equal to the current file's domain. It's not clear to me why this is no longer valid for `homeassistant` ?

Reply:
Especially since `homeassistant` is the main driver for this check to begin with. Other intents are not tested in other domains than the one that defines them.